articles/openstack/ostack-how-basic-heat-template.md

```diff
@@ -0,0 +1,164 @@
+      instance_uuid: { get_resource: heat_server }
+  heat_network:
+    type: OS::Neutron::Net
+    properties:
+      admin_state_up: true
+      name: heat_network
+  heat_network_subnet:
+    type: OS::Neutron::Subnet
+    properties:
+      network: { get_resource: heat_network }
+      cidr: "10.1.1.0/24"
+      dns_nameservers: ["8.8.8.8"]
+      gateway_ip: "10.1.1.1"
+      ip_version: 4
+  heat_router:
+    type: OS::Neutron::Router
+    properties:
+      external_gateway_info: { network: internet }
+      name: heat_router
+  heat_router_interface:
+    type: OS::Neutron::RouterInterface
+    properties:
+      router_id: { get_resource: heat_router }
+      subnet: { get_resource: heat_network_subnet }
+  heat_server_port:
+    type: OS::Neutron::Port
+    properties:
+      network: { get_resource: heat_network }
+      fixed_ips:
+        - subnet_id: { get_resource: heat_network_subnet }
+  heat_server:
+    type: OS::Nova::Server
+    properties:
+      name: heat_server
+      flavor: { get_param: flavor }
+      image: { get_param: image }
+      networks:
+        - port: { get_resource: heat_server_port}
+  heat_server_public_ip:
+    type: OS::Neutron::FloatingIP
+    properties:
+      floating_network: "internet"
+  heat_server_ip_assoc:
+    type: OS::Neutron::FloatingIPAssociation
+    properties:
+      floatingip_id: { get_resource: heat_server_public_ip }
+      port_id: { get_resource: heat_server_port }
+
+outputs:
+  heat_server_public_ip:
+    description: IP Address of the deployed heat_server instance
+    value: { get_attr: [ heat_server_public_ip, floating_ip_address ]}   
+```
+
+## Deploying the stack
+
+> [!NOTE]
+> You will need to source your RC file before you run the commands below.
+
+1. Create a stack by running the following command in the OpenStack CLI:
+
+    ```
+    $ openstack stack create -t basic-stack.yaml basic-stack
+    ```
+    SYNTAX:
+    ```
+    openstack stack create -t  <template name> <stack name>
+    ```
+
+    This command will return the following:
+    
+    ```
+    +---------------------+--------------------------------------+
+    | Field               | Value                                |
+    +---------------------+--------------------------------------+
+    | id                  | 6be269ec-d22f-4cf0-bf04-df71cc6dcf75 |
+    | stack_name          | basic-stack                          |
+    | description         | No description                       |
+    | creation_time       | 2019-02-12T15:58:20Z                 |
+    | updated_time        | None                                 |
+    | stack_status        | CREATE_IN_PROGRESS                   |
+    | stack_status_reason | Stack CREATE started                 |
+    +---------------------+--------------------------------------+
+    ```
+
+    
+## Next steps
+
+### Generate a heat template for your existing environment.
+Flame a standalone tool to convert from an existing environment into a heat template.
```

Comment:
Suggest:
If you want to create template from your existing environment, you can use a tool like Flame that converts an existing environment into a Heat template.